### ISSUE TITLE
Performance gap in KITTI validation and test set

### ISSUE THREAD

**Issue by JCZ404** [KuanchihHuang/MonoDTR#12]
Hi, Thanks for your great work and this awesome released code. But I encountered some problems to reproduce the performance on KITTI test set. I used the provided code to train with KITTI train-split(3712 images), and I got the **best performance on KITTI validation set during training** is `23.93, 18.12, 15.04`, which is similar with the reported  performance in your paper, i.e. `24.53, 18.57, 15.51`, but there is still ~0.5 performance gap across all the difficulty level. I also train the model with KITTI train + val split(3712 + 3769 images), and submit the inference results on test set(7518 images) with the final model to the KITTI online server, and finally, I got the performance with `20.59 14.93 12.80`, which is close to the reported results `21.99 15.39 12.73`, but there are still about ~1.4 and ~0.5 performance gaps on Easy and Moderate level, respectively. Is there results reasonable? Or is there any detail I missed? Could you provide the checkpoint file which generate the reported performance on KITTI test in your paper? Thanks a lot!
![image](https://user-images.githubusercontent.com/51013927/232960371-d2b4f091-caa2-42bc-8028-d883d8e6ee7e.png)

**Comment by shipman624** [comment]
Hi, I have a similar problem as you. May I ask you if you took the last epoch of training and submitted it for training? Also I would like to ask you how you generate the inference results, I suspect that I am not doing it correctly.

**Comment by ksh11023** [comment]
@JCZ404

Hi, 
Can I ask you a question regarding training on trainval and summiting to the official KITTI online server?

In what format do you summit the results ? Is it this txt file?

<img width="117" alt="스크린샷 2023-06-05 오후 1 38 06" src="https://github.com/KuanchihHuang/MonoDTR/assets/50094436/04458136-09b3-4bbd-9a9a-b2fb5a6bd6b8">


Thank you.
